src/background/tab-manager.ts

```diff
@@ -155,15 +170,19 @@ export default class TabManager {
 
     async sendMessage(getMessage: (url: string, frameUrl: string) => any) {
         (await queryTabs({}))
-            .filter((tab) => this.ports.has(tab.id))
+            .filter((tab) => this.tabs.has(tab.id))
             .forEach((tab) => {
-                const framesPorts = this.ports.get(tab.id);
-                framesPorts.forEach(({url, port}, frameId) => {
+                const frames = this.tabs.get(tab.id);
+                frames.forEach(({url, state}, frameId) => {
+                    if (state === 'frozen') {
+                        // TODO: avoid sending messages to frozen tabs for performance reasons.
+                        logInfo('Sending message to a frozen tab.');
```

Comment:
Hmmm, what if the frame resumed, will it stay white? Should something be sent to it?

Reply:
With this patch, this code will just log a message about tab/frame being frozen and then will continue and send the message anyway. This behavior matches exactly what extension does before this patch. 

Also, this is most likely suboptimal (because it wakes up the frozen frame), so I left a comment and plan to file a follow-up PR.

After this patch, there are still two issues which will require follow-up patches:
 - Imagine a frame was frozen (which background knows about because it receives a message) and then unloaded (which happens without any callbacks for performance reasons) and background tries to send a message to it. Can we determine if frame dies like that or just ignore existence of such "ghost" frames?
 - Avoid sending excessive messages to frozen tabs all at once. E.g., imagine I have lots of frozen tabs folded to tray or behind other windows and then I toggle global on/off switch. Background probably should keep a record of messages addressed to frozen frames and then actually deliver them once frame wakes up.

Reply:
Maybe there should be some error, if message is sent to a not existent frame.

Great, looking forward to other patches.